src/components/SearchInput/Input.js

```diff
@@ -48,9 +48,13 @@ function Input({ onRefresh, placeholder, query }) {
       // API. On all other pages that this component is used, onRefresh is
       // expected to be null
       if (typeof onRefresh === 'function') {
-        router.push(`/search?q=${searchTerm}`, `/search?q=${searchTerm}`, {
-          shallow: true
-        });
+        router.push(
+          `/search?q=${searchTerm}&lang=${language}`,
+          `/search?q=${searchTerm}&lang=${language}`,
```

Comment:
```suggestion
          undefined,
```

Since they're the same, `undefined` should do the trick: https://nextjs.org/docs/api-reference/next/router#routerpush